http://sourceware.org/ml/libc-alpha/2011-06/msg00006.html

From 57912a71cc379f8582e3ea272782dc39c6c02e45 Mon Sep 17 00:00:00 2001
From: Ulrich Drepper <drepper@gmail.com>
Date: Wed, 15 Jun 2011 22:43:58 -0400
Subject: [PATCH] Fix typo in recent resolver change which causes segvs

---
 ChangeLog         |    5 +++++
 resolv/res_send.c |    4 ++--
 2 files changed, 7 insertions(+), 2 deletions(-)

2011-06-15  Ulrich Drepper  <drepper@gmail.com>

	* resolv/res_send.c (__libc_res_nsend): Fix typos in last patch.  We
	need to dereference resplen2.

diff --git a/resolv/res_send.c b/resolv/res_send.c
index 97142b7..a001c1e 100644
--- a/resolv/res_send.c
+++ b/resolv/res_send.c
@@ -549,7 +549,7 @@ __libc_res_nsend(res_state statp, const u_char *buf, int buflen,
 				    ns, ansp, ansp2, nansp2, resplen2);
 			if (n < 0)
 				return (-1);
-			if (n == 0 && (buf2 == NULL || resplen2 == 0))
+			if (n == 0 && (buf2 == NULL || *resplen2 == 0))
 				goto next_ns;
 		} else {
 			/* Use datagrams. */
@@ -559,7 +559,7 @@ __libc_res_nsend(res_state statp, const u_char *buf, int buflen,
 				    ansp2, nansp2, resplen2);
 			if (n < 0)
 				return (-1);
-			if (n == 0 && (buf2 == NULL || resplen2 == 0))
+			if (n == 0 && (buf2 == NULL || *resplen2 == 0))
 				goto next_ns;
 			if (v_circuit)
 			  // XXX Check whether both requests failed or
-- 
1.7.5.3
